### PR TITLE
More cleanup and improvements enabled by signature move

### PR DESCRIPTION
**Pull request by chamons**
- Make ComputeSignatures an internal idea, triggered by property changes that invalidate signature

### CODE REVIEW

**Comment by chamons** [comment]
Mostly wraps up https://github.com/mono/Embeddinator-4000/issues/276

File: objcgen/processedtypes.cs [hunk 1]
```diff
@@ -154,11 +148,36 @@ public enum MethodType {
 		DefaultValueWrapper,
 	}
 
-	public class ProcessedMethod : ProcessedMemberBase {
+	public abstract class ProcessedMemberWithParams : ProcessedMemberBase {
```
Comment (spouliot) [inline objcgen/processedtypes.cs]: `Params` -> `Parameters` 
no need for shortcuts as we have code completion ;-)

File: objcgen/processedtypes.cs [hunk 2]
```diff
@@ -154,11 +148,36 @@ public enum MethodType {
...
+			get
+			{
+				return firstDefaultParameter;
+			}
+			set
+			{
+				firstDefaultParameter = value;
+				ComputeSignatures ();
+			}
+		}
```
Comment (spouliot) [inline objcgen/processedtypes.cs]: formatting ^ of `{` to mono guidelines

**Comment by chamons** [comment]
Fixed CppSharp bump.

File: objcgen/objcgenerator.cs [hunk 3]
```diff
@@ -993,9 +989,8 @@ void GenerateDefaultValuesWrapper (ProcessedMemberBase member)
 
 			MethodBase mb = method != null ? (MethodBase)method.Method : ctor.Constructor;
 			MethodInfo mi = mb as MethodInfo;
-			string objcsig;
-			string monosig;
-			var parameters = method != null ? method.Method.GetParameters () : ctor.Constructor.GetParameters ();
+			int firstDefaultParameter = method != null ? method.FirstDefaultParameter : ctor.FirstDefaultParameter;
```
Comment (spouliot) [inline objcgen/objcgenerator.cs]: `FirstDefaultParameter` was moved to a shared base type, can't we take advantage of this ?

File: objcgen/objcgenerator.cs [hunk 4]
```diff
@@ -993,9 +989,8 @@ void GenerateDefaultValuesWrapper (ProcessedMemberBase member)
...
+			var parameters = method != null ? method.Parameters : ctor.Parameters;
```
Comment (spouliot) [inline objcgen/objcgenerator.cs]: same here for `Parameters`

File: objcgen/processedtypes.cs [hunk 5]
```diff
@@ -162,19 +162,17 @@ public ProcessedMemberWithParams (Processor processor) : base (processor)
 		int firstDefaultParameter;
 		public int FirstDefaultParameter
 		{
```
Comment (spouliot) [inline objcgen/processedtypes.cs]: ^ missed one ;-)

**Pull request merged**